File: app-Sample/src/main/java/com/sjbt/sdk/sample/utils/AssetUtils.java
package com.sjbt.sdk.sample.utils;

import android.content.Context;
import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import java.io.IOException;
import java.io.InputStream;

public class AssetUtils {

    /**
     * 获取Asset文件-bin文件
     */
    public static byte[] getAssetBytes(Context context, String fileName) {
        byte[] buffer = null;
        try {
            InputStream is = context.getAssets().open(fileName);
            int size = is.available();
            buffer = new byte[size];
            is.read(buffer);
            is.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return buffer;
    }

    /**
     * 获取Asset文件-图片
     */
    public static Bitmap getAssetBitmap(Context context, String fileName) {
        Bitmap bitmap = null;
        AssetManager assetManager = context.getAssets();
        try {
            InputStream inputStream = assetManager.open(fileName);//filename是assets目录下的图片名
            bitmap = BitmapFactory.decodeStream(inputStream);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return bitmap;
    }

    // 从 Assets 文件夹中获取 json 文件内容
    public static String getJSONFromAsset(Context context, String fileName) {
        String json = null;
        try {
            InputStream is = context.getAssets().open("cloudDial/" + fileName);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }
    // 从 Assets 文件夹中获取 json 文件内容
    public static String getJSONFromAssetDeviceModel(Context context) {
        String json = null;
        try {
            InputStream is = context.getAssets().open("DeviceInfo.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }

    public static String getJSONFromAssetRoot(Context context, String fileName) {
        String json = null;
        try {
            InputStream is = context.getAssets().open(fileName);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }
}

File: lib-chuanyin/lib-sj-sdk/src/main/java/com/sjbt/sdk/utils/AssetUtils.kt
package com.sjbt.sdk.utils

import android.content.Context
import android.util.Log
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStream
import java.io.UnsupportedEncodingException
import java.util.*

fun readSportTypeJsonFromAssets(context: Context): String? {
    val assetManager = context.assets
    var istream: InputStream? = null
    var jsonData = ""

    try {
        istream = assetManager.open("sports_data.json")
        val reader = Scanner(istream, "UTF-8")
        while (reader.hasNextLine()) {
            jsonData += reader.nextLine() + "\n"
        }
    } catch (e: FileNotFoundException) {
        Log.e("JSON_FILE", "File not found", e)
    } catch (e: UnsupportedEncodingException) {
        Log.e("JSON_FILE", "Unsupported Encoding", e)
    } finally {
        try {
            istream?.close()
        } catch (e: IOException) {
            Log.e("JSON_FILE", "Error in closing the stream", e)
        }
    }
    return jsonData
}